components/eamxx/src/physics/mam/mam_coupling.hpp

```diff
@@ -592,21 +592,26 @@ void compute_vertical_layer_heights(const Team& team,
   EKAT_KERNEL_ASSERT_MSG(column_index == team.league_rank(),
     "Given column index does not correspond to given team!");
 
+  //outputs
   const auto dz = ekat::subview(dry_atm.dz, column_index);
   const auto z_iface  = ekat::subview(dry_atm.z_iface, column_index);
   const auto z_mid    = ekat::subview(dry_atm.z_mid, column_index);
+  //inputs
   const auto qv = ekat::subview(dry_atm.qv, column_index);
   const auto p_mid = ekat::subview(dry_atm.p_mid, column_index);
   const auto T_mid = ekat::subview(dry_atm.T_mid, column_index);
   const auto pseudo_density = ekat::subview(dry_atm.p_del, column_index);
+  
   // NOTE: we are using dry qv. Does calculate_dz require dry or wet?
   PF::calculate_dz(team, pseudo_density, p_mid, T_mid, qv, // inputs
             dz);//output
   team.team_barrier();
+  EKAT_KERNEL_ASSERT_MSG(dry_atm.z_surf == 0, "dry_atm.z_surf must be zero");
```

Comment:
I don't know anything about mam, or what z_surf represents, but...does this mean you're assuming no topography?

Reply:
I mimicked this from EAM but @kaizhangpnl might know more about it. [Here ](https://github.com/eagles-project/scream/blob/3005a4b8ef245c06f0a769a87c3c85bf0c29e519/components/eam/src/physics/cam/geopotential.F90#L83)is the EAM code from where I assumed that it should be zero

Reply:
It depends where it comes from. Grepping in the mam folder, it seems it is always set to zero (and never changed):
```bash
$ git grep z_surf
eamxx_mam_aci_process_interface.cpp:  dry_atm_.z_surf = 0;
eamxx_mam_dry_deposition_process_interface.cpp:  dry_atm_.z_surf    = 0.0;  // FIXME: for now
eamxx_mam_microphysics_process_interface.cpp:  dry_atm_.z_surf = 0.0; // FIXME: for now
eamxx_mam_optics_process_interface.cpp:  dry_atm_.z_surf    = 0.0;
mam_coupling.hpp:  Real          z_surf;    // height of bottom of atmosphere [m]
mam_coupling.hpp:  EKAT_KERNEL_ASSERT_MSG(dry_atm.z_surf == 0, "dry_atm.z_surf must be zero");
mam_coupling.hpp:  PF::calculate_z_int(team, mam4::nlev, dz, dry_atm.z_surf, //inputs
```
But the comment "FIXME: for now" makes me think this is just temporary.

Reply:
I should remove the `FIXME` comments for `z_surf`. I just grepped the EAMxx code and it is set to zero at all places there as well (e.g. SHOC, TMS, surface coupling, etc.). So it should be zero everywhere.

Reply:
state%zi in EAM is the geopotential height (at the layer interface) above the surface (not above the sea level). At the bottom of the model, it is zero by definition. On the other hand, I am not sure how z_surf is used exactly in MAMxx (just to calculate dz?).

Reply:
You should not need z_surf to compute dz though...

Reply:
By searching "z_surf" in the repo, it seems that it has different definitions in different parameterizations: 

https://github.com/search?q=repo%3AE3SM-Project%2Fscream%20z_surf&type=code 

Maybe it (how it is defined and used) needs to be described more clearly in the code.

Reply:
In MAM4xx codes, we use `z_surf` to compute `z_iface` (or `z_int` or state%zi in EAM or z interface values). We followed EAMxx to compute `z_iface` using the `PF::calculate_z_int` function. In [EAM](https://github.com/eagles-project/scream/blob/3005a4b8ef245c06f0a769a87c3c85bf0c29e519/components/eam/src/physics/cam/geopotential.F90#L83), we also set it to zero at the surface.